pkg/catalog/install.go

```diff
@@ -0,0 +1,112 @@
+package catalog
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kjson "k8s.io/apimachinery/pkg/runtime/serializer/json"
+)
+
+// Writer takes a profile and writes it out into a medium.
+type Writer interface {
+	Output(prof *profilesv1.ProfileSubscription) error
+}
+
+// FileWriter is a Writer using a file as backing medium.
+type FileWriter struct {
+	Filename string
+}
+
+// Output writes the profile subscription yaml data into a given file.
+func (fw *FileWriter) Output(prof *profilesv1.ProfileSubscription) error {
+	e := kjson.NewSerializerWithOptions(kjson.DefaultMetaFactory, nil, nil, kjson.SerializerOptions{Yaml: true, Strict: true})
+	f, err := os.OpenFile(fw.Filename, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
```

Comment:
might `0600` be a bit tight?

Reply:
Hmm, good point. I guess 0644 is good enough too.